src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -471,5 +471,83 @@ private static unsafe void Populate83FileNameFromRandomBytes(byte* bytes, int by
             chars[10] = s_base32Char[(bytes[6] & 0x1F)];
             chars[11] = s_base32Char[(bytes[7] & 0x1F)];
         }
+
+        /// <summary>
+        /// Create a relative path from one path to another. Paths will be resolved before calculating the difference.
+        /// Default path comparison for the active platform will be used (OrdinalIgnoreCase for Windows, Ordinal for Unix).
+        /// </summary>
+        /// <param name="relativeTo">The source path the output should be relative to. This path is always considered to be a directory.</param>
+        /// <param name="path">The destination path.</param>
+        /// <returns>The relative path or <paramref name="path"/> if the paths don't share the same root.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="relativeTo"/> or <paramref name="path"/> is <c>null</c> or an empty string.</exception>
+        public static string GetRelativePath(string relativeTo, string path)
+        {
+            return GetRelativePath(relativeTo, path, PathInternal.DefaultPathComparison);
+        }
+
+        private static string GetRelativePath(string relativeTo, string path, StringComparison comparisonType)
+        {
+            if (string.IsNullOrEmpty(relativeTo)) throw new ArgumentNullException(nameof(relativeTo));
+            if (string.IsNullOrWhiteSpace(path)) throw new ArgumentNullException(nameof(path));
+            Debug.Assert(comparisonType == StringComparison.Ordinal || comparisonType == StringComparison.OrdinalIgnoreCase);
+
+            relativeTo = GetFullPath(relativeTo);
+            path = GetFullPath(path);
+
+            // Need to check if the roots are different- if they are we need to return the "to" path.
+            if (!PathInternal.AreRootsEqual(relativeTo, path, comparisonType))
+                return path;
+
+            int commonLength = PathInternal.GetCommonPathLength(relativeTo, path, ignoreCase: comparisonType == StringComparison.OrdinalIgnoreCase);
+
+            // If there is nothing in common they can't share the same root, return the "to" path as is.
+            if (commonLength == 0)
+                return path;
+
+            // Trailing separators aren't significant for comparison
+            bool pathEndsInSeparator = PathInternal.EndsInDirectorySeparator(path);
+
+            int relativeToLength = relativeTo.Length - (PathInternal.EndsInDirectorySeparator(relativeTo) ? 1 : 0);
+            int pathLength = path.Length - (pathEndsInSeparator ? 1 : 0);
```

Comment:
Nit: it's minor, but I'd prefer it if these didn't do unnecessary arithmetic, e.g.
```C#
int pathLength = path.Length;
if (pathEndsInSeparator)
{
    pathLength--;
}
```